### PR TITLE
add `red-hat-storage/mcg-osd-deployer`

### PR DESCRIPTION
**Pull request by rexagod**
added CI support for the `red-hat-storage/mcg-osd-deployer` github
repository.

Signed-off-by: Pranshu Srivastava <rexagod@gmail.com>

### CODE REVIEW

**Comment by rexagod** [comment]
@BlaineEXE this is good to go now, PTAL, thanks.

**Comment by rexagod** [comment]
@bbguimaraes Please review.

File: ci-operator/config/red-hat-storage/mcg-osd-deployer/red-hat-storage-mcg-osd-deployer-main.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  steps:
+    cluster_profile: aws
+    test:
+    - as: lint-aws
+      commands: make lint
+      from: src
+      resources:
+        requests:
+          cpu: 100m
+          memory: 200Mi
```
Comment (bbguimaraes) [inline ci-operator/config/red-hat-storage/mcg-osd-deployer/red-hat-storage-mcg-osd-deployer-main.yaml:53]: Are these right?  Maybe you need a test cluster for your `make test`, but I highly doubt you need one for coverage or lint tests.

**Review by bbguimaraes — CHANGES_REQUESTED** [review]

File: ci-operator/config/red-hat-storage/mcg-osd-deployer/red-hat-storage-mcg-osd-deployer-main.yaml (hunk 1 above)
Comment (rexagod) [reply ci-operator/config/red-hat-storage/mcg-osd-deployer/red-hat-storage-mcg-osd-deployer-main.yaml:53]: Makes sense. Do you mean I should use `USE_EXISTING_CLUSTER=1` for both of these commands then?

Comment (bbguimaraes) [reply ci-operator/config/red-hat-storage/mcg-osd-deployer/red-hat-storage-mcg-osd-deployer-main.yaml:53]: I don't know how your tests are structured, but `cluster_profile` is (part of) what is used to request that a temporary cluster be created for every execution of your test (see https://docs.ci.openshift.org/docs/architecture/step-registry/#ci-operator-test-configuration).

You certainly don't need one just to execute a `lint` program over your source code.  As they're written, your tests do not fully request one and instead just take one spot from the shared pool of cluster unnecessarily.

**Comment by rexagod** [comment]
/hold
Will make `lint` use existing cluster and move `test-coverage` under `test-suite` itself.

File: ci-operator/config/red-hat-storage/mcg-osd-deployer/red-hat-storage-mcg-osd-deployer-main.yaml (hunk 1 above)
Comment (rexagod) [reply ci-operator/config/red-hat-storage/mcg-osd-deployer/red-hat-storage-mcg-osd-deployer-main.yaml]: Using existing cluster to run `lint` step now, and merged `test-coverage` into `test-suite`, so just two steps now: https://github.com/red-hat-storage/mcg-osd-deployer/pull/30

@bbguimaraes Is this okay?

**Review by jarrpa — APPROVED** [review]
/lgtm

**Pull request merged**